### PR TITLE
TCMN-68: Improve regular expressions

### PR DESCRIPTION
**Pull request by mitogh**
Removal of non required characters and groups.

Ticket: [TCMN-68]

[TCMN-68]: https://moderntribe.atlassian.net/browse/TCMN-68

### CODE REVIEW

File: src/Tribe/Timezones.php [hunk 1]
```diff
@@ -615,7 +615,7 @@ public static function get_valid_timezone( $timezone_candidate ) {
 			return $timezone_candidate->getName();
 		}
 
-		$timezone_string = preg_replace( '/(\\+||\\-)0$/', '', $timezone_candidate );
+		$timezone_string = preg_replace( '/[+\-]0$/', '', $timezone_candidate );
```
Comment (sc0ttkclark) [inline src/Tribe/Timezones.php:618]:
```suggestion
		$timezone_string = preg_replace( '/[+-]0$/', '', $timezone_candidate );
```

Dash at end of range doesn't have to be escaped: https://stackoverflow.com/questions/30398445/matching-dot-in-a-range-preg-match

File: src/Tribe/Timezones.php [hunk 2]
```diff
@@ -408,7 +408,7 @@ public static function is_valid_timezone( $candidate ) {
 	 */
 	public static function timezone_from_utc_offset( $utc_offset_string ) {
 		// Test for strings looking like "UTC-2" or "UTC+5.25" etc
-		if ( ! preg_match( '/^UTC[\-\+]{1}[0-9\.]{1,4}$/', $utc_offset_string ) ) {
+		if ( ! preg_match( '/^UTC[\-+][0-9.]{1,4}$/', $utc_offset_string ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Timezones.php:411]:
```suggestion
		if ( ! preg_match( '/^UTC[+-][0-9.]{1,4}$/', $utc_offset_string ) ) {
```

Dash at end of range doesn't have to be escaped: https://stackoverflow.com/questions/30398445/matching-dot-in-a-range-preg-match

**Comment by bordoni** [comment]
@mitogh can you add a changelog to this one? I feel like including this on B20.07 would be ideal.

**Comment by mitogh** [comment]
Sure, let me take care of that.

Thanks @bordoni

**Comment by mitogh** [comment]
> Can we get testing coverage for this?

Oh this was cover already on the past, the fact that tests are not failing is an indicator that the PR does not impact those tests.

- https://github.com/moderntribe/tribe-common/blob/master/tests/wpunit/Tribe/TimezonesTest.php#L126-L167

**Comment by mitogh** [comment]
I think we need an approval of @sc0ttkclark and @Camwyn in order to move this PR forward, let me know in case you have additional feedback.

Thanks.

**Comment by Camwyn** [comment]
Don't let me stop you!

File: readme.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@
 = [TBD] TBD =
 
 * Fix - Added IE11 compatibility for the toggles styles using `tribe-common-form-control-toggle` CSS class. [ET-865]
+* Tweak - Improve regular expressions to parse UTC timezones by removing non required grouping and characters [TCMN-68]
```
Comment (sc0ttkclark) [inline readme.txt:8]:
```suggestion
* Tweak - Improve regular expressions used to parse UTC timezones by removing non-required grouping and characters. [TCMN-68]
```

**Pull request merged**